my_list.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'my_enumerable_module'
+class Mylist
+  include MyEnumerable
+
+  def initialize(*list)
+    @list = list
+  end
+
+  def each(&block)
+    @list.each(&block)
+  end
```

Comment:
- [ ] Kindly make sure to implement this method to yield list members from the list. You've done so but using `each`. While this is correct, the purpose of this method is to learn about `yield` and how it works in a method. You can do something like the following to replicate the `each` behavior without actually using it.

```ruby
def each
    return to_enum(:each) unless block_given?

    counter = 0
    while counter < @list.length
      yield(@list[counter])
      counter += 1
    end
    @list
end
```

Reply:
@ShadyShawkat  Thank you so much for such an insightful review, We will implement the changes as recommended.